### PR TITLE
TT 36 group creation

### PR DESCRIPTION
**Pull request by galen94**

### CODE REVIEW

File: services/group-service.ts [hunk 1]
```diff
@@ -37,18 +37,24 @@ export default class GroupService implements GroupServiceI{
...
+        if(creatorUid === "me" || !req.body.users){
             res.sendStatus(503);
             return;
         }
+        req.body.users.push(creatorUid);
+
         try {
-            GroupService.groupDao.createGroup(creatorUid, allUids, req.body)
+            const existingGroups = await GroupService.groupDao.findAllUserGroups(creatorUid);
+            const filteredGroups = existingGroups.filter(group => group.name.toLowerCase() === req.body.group.name.toLowerCase());
```
Comment (sghosalkar) [inline services/group-service.ts]: Let's discuss this again. I thought this and later realized that we are making a mistake here.

Comment (sghosalkar) [inline services/group-service.ts]: Let's discuss this again. I thought this and later realized that we are making a mistake here.

File: interfaces/group-dao-I.ts [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+import User from "../models/users/user";
 
 /**
  * @file Declares API for Group related data access object methods
  */
 export default interface GroupDaoI {
     findAllUserGroups (uid: string): Promise<Group[]>;
-    createGroup (creatorUid: string, uids: string[], group: Group): Promise<Group>;
+    createGroup (creatorUid: string, users: User[], group: Group): Promise<Group>;
     deleteGroup (gid: string): Promise<any>;
```
Comment (sghosalkar) [inline interfaces/group-dao-I.ts:9]: I think we should keep this as `uids: string[]` and not user objects

Comment (galen94) [reply interfaces/group-dao-I.ts:9]: Yeah it wasn't working, so I had to change it.

File: daos/group-dao.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@
 import GroupDaoI from "../interfaces/group-dao-I";
 import GroupModel from "../mongoose/messages/group-model";
 import Group from "../models/messages/group";
+import User from "../models/users/user";
```
Comment (sghosalkar) [inline daos/group-dao.ts]: Looks like this import is unused

File: interfaces/group-dao-I.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import Group from "../models/messages/group";
+import User from "../models/users/user";
```
Comment (sghosalkar) [inline interfaces/group-dao-I.ts]: Unused import, can be removed

**Review by sghosalkar — APPROVED** [review]

**Review by tomirisk — APPROVED** [review]
Good job! It runs on my end smoothly.

**Pull request merged**